Duplicati/Library/Main/Operation/Backup/FileEnumerationProcess.cs

```diff
@@ -386,6 +392,45 @@ private static bool AttributeFilter(string path, FileAttributes attributes, Snap
             // All the way through, yes!
             return true;
         }
+
+        /// <summary>
+        /// Trailing characters that are invalid in Windows.
+        /// </summary>
+        private static readonly string[] InvalidTrailingWindowsPathCharacters =
+        {
+            ".",
+            " "
+        };
+
+        /// <summary>
+        /// Trailing characters that are invalid in Windows with the
+        /// directory separator appended.
+        /// </summary>
+        private static readonly string[] InvalidTrailingWindowsPathCharactersWithSeparator =
+            InvalidTrailingWindowsPathCharacters.Select(Util.AppendDirSeparator).ToArray();
+
+        /// <summary>
+        /// Tests if path is invalid for Windows.
+        /// </summary>
+        /// <remarks>
+        /// For example, calling Path.GetFullPath() in Windows on files or
+        /// directories ending in a dot or a space will return a path without
+        /// the dot or the space.
+        /// </remarks>
+        /// <param name="path">The current path.</param>
+        /// <param name="attributes">The file or folder attributes.</param>
+        /// <returns>Returns true if file or directory ends in an invalid character for Windows.</returns>
+        private static bool IsInvalidWindowsPath(string path, FileAttributes attributes)
+        {
+            if ((attributes & FileAttributes.Directory) != 0)

```

Comment:
Can we simplify this by using `attributes.HasFlag(FileAttributes.Directory)`?

Reply:
Yes.  I've pushed a change to use `HasFlag()`, thanks.